### PR TITLE
Location Reassignment Phase 2

### PR DESCRIPTION
**Pull request by mkangia**
[Spec](https://docs.google.com/document/d/1h6kTr9M0FumswP3-gr_84OLeUHRmn5Y10dCpSAS3Y9g/edit#heading=h.kt0o2e3g01po)
(Built picking from approach check in https://github.com/dimagi/commcare-hq/pull/26898)

##### SUMMARY
This implements the core part of the second phase of Location Reassignment where locations are archived and users are deactivated. This currently assumes that locations are already present in the system. 
As a follow up PR i would cater the case where we create locations on the fly.
But want to get the core update logic in place.

##### FEATURE FLAG
`LOCATION_REASSIGNMENT`

##### PRODUCT DESCRIPTION
This hooks in the UI and just adds a checkbox to actually do updates.

### CODE REVIEW

File: custom/icds/location_reassignment/parser.py [hunk 1]
```diff
@@ -123,3 +130,27 @@ def _validate_descendants_archived(self):
...
+
+    def _perform_transitions(self, operation, transitions):
+        for old_site_codes, new_site_codes in transitions.items():
+            deprecate_locations(operation, self._get_locations(old_site_codes),
+                                self._get_locations(new_site_codes))
+
+    def _get_locations(self, site_codes):
+        site_codes = site_codes if isinstance(site_codes, list) else [site_codes]
+        locations = [self.locations_by_site_code[site_code] for site_code in site_codes]
+        assert len(locations), len(site_codes)
```
Comment (orangejenny) [inline custom/icds/location_reassignment/parser.py]: Do you mean for `len(site_codes)` to be the assertion failure message? Seems more useful to use something like "Expected locations but found none."

**Review by orangejenny — APPROVED** [review]

File: custom/icds/location_reassignment/parser.py (hunk 1 above)
Comment (mkangia) [reply custom/icds/location_reassignment/parser.py]: valid point, updated to raise exception
https://github.com/dimagi/commcare-hq/pull/26983/commits/8d11300ef844b44e80571e4b0bd6d14da71ac849

**Comment by mkangia** [comment]
@orangejenny i think i am done with changes to this PR and would like to merge it to the main one now. Let me know if you have any feedback on changes post last review.

**Review by orangejenny — APPROVED** [review]
Looks good to me. Is the email notification going into a future PR?

**Comment by mkangia** [comment]
@orangejenny yes, unless the first PR gets delayed due to review then i would include it

**Pull request merged**